server/region_handler.go

```diff
@@ -40,13 +44,74 @@ const (
 	pDBName    = "db"
 	pTableName = "table"
 	pRegionID  = "regionID"
+	pRecordID  = "recordID"
+	pStartTS   = "startTS"
 )
 
 const (
 	headerContentType = "Content-Type"
 	contentTypeJSON   = "application/json"
 )
 
+type kvStore interface {
+	GetRegionCache() *tikv.RegionCache
+	SendReq(bo *tikv.Backoffer, req *tikvrpc.Request, regionID tikv.RegionVerID, timeout time.Duration) (*tikvrpc.Response, error)
+}
+
+type regionHandlerTool struct {
+	bo          *tikv.Backoffer
```

Comment:
What do you mean?The `regionHanderTool` is used to generate and store common tools for each `regionHander`.  I set the `backoffer` as a member of the `tool` since more than one regionHander needed it @tiancaiamao